### ISSUE TITLE
Doesn't work with dot in config file name.

### ISSUE THREAD

**Issue by whmountains** [mehcode/config-rs#101]
I get `configuration file "default.settings" not found` when I try to load a file called "default.settings.toml".  The code looks like this:

```rust
let mut settings = config::Config::default();
settings
    .merge(config::File::with_name("default.settings")).unwrap();
```

**Comment by kunicmarko20** [comment]
I didn't dig too deep but _probably_ if you have a dot without the real extension it recognizes it as the file extension and doesn't apply the one you use. In this case, it thinks your extension is `.settings` so it doesn't add `.toml` to it in this method: https://github.com/mehcode/config-rs/blob/master/src/file/source/file.rs#L27. If you explicitly set `default.settings.toml` it works.

**Comment by kunicmarko20** [comment]
https://github.com/mehcode/config-rs/blob/master/src/file/source/file.rs#L33 here it will return false since the file `./default.settings` doesn't exist.

Now we are here https://github.com/mehcode/config-rs/blob/master/src/file/source/file.rs#L60,
it tries to apply known extensions but since your extension is `.settings ` it basically replaces it, creating `default.toml` and that again doesn't exists, in the end it will just fail here  https://github.com/mehcode/config-rs/blob/master/src/file/source/file.rs#L80

So basically, this problem is in the `PathBuf` and not in this library. Here is an example:

https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=cc8f94f432b931c7bf7dac75448f835c

**Comment by matthiasbeyer** [comment]
Is this still an issue? If yes, can you provide patches?

**Comment by CfirTsabari** [comment]
i fixed it in #306

**Comment by CfirTsabari** [comment]
@matthiasbeyer can we get a new version with this fix? It didn't get into 0.13.0

**Comment by matthiasbeyer** [comment]
Yep: https://github.com/mehcode/config-rs/pull/313
